src/azure-cli/azure/cli/command_modules/acs/_validators.py

```diff
@@ -852,3 +853,12 @@ def validate_custom_ca_trust_certificates(namespace):
         if hasattr(namespace, 'os_type') and namespace.os_type != "Linux":
             raise ArgumentUsageError(
                 '--custom-ca-trust-certificates can only be set for linux nodepools')
+
+
+def validate_gateway_prefix_size(namespace):
+    """Validates the gateway prefix size."""
+    if namespace.gateway_prefix_size is not None:
+        if not hasattr(namespace, 'mode') or namespace.mode != CONST_NODEPOOL_MODE_GATEWAY:
+            raise ArgumentUsageError("--gateway-prefix-size can only be set for Gateway-mode nodepools")
+        if namespace.gateway_prefix_size < 28 or namespace.gateway_prefix_size > 31:
+            raise CLIError("--gateway-prefix-size must be in the range [28, 31]")
```

Comment:
Please use the specific error type `InvalidArgumentValueError` instead of `CLIError`
```suggestion
            raise InvalidArgumentValueError("--gateway-prefix-size must be in the range [28, 31]")
```